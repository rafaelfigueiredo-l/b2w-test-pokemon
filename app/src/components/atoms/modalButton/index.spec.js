import ModalButton from "./";

describe("<ModalButton />", () => {
  const onChangeMock = jest.fn();

  const component = <ModalButton onChange={onChangeMock()}>Teste</ModalButton>;

  const mounted = mount(component);
  const wrapper = shallow(component);

  it("should mount correctly ModalButton component", () => {
    expect(mounted).toMatchSnapshot();
  });

  it("should unmount correctly ModalButton component", () => {
    expect(mounted.unmount()).toEqual({});
  });

  it("should have className ModalButton", () => {
    expect(wrapper.find(".modalButton")).toHaveLength(1);
  });

  it("should call onChange function", () => {
    expect(onChangeMock).toHaveBeenCalledTimes(1);
    expect(onChangeMock).toBeCalled();
  });

  it("should have a children", () => {
    expect(wrapper.find(".modalButton").children()).toHaveLength(1);
  });
});
