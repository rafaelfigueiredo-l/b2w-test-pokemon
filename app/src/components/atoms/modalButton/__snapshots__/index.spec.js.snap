// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ModalButton /> should mount correctly ModalButton component 1`] = `
<styled.button
  className="modalButton"
>
  <button
    className="sc-AxjAm lhOTJ modalButton"
  >
    Teste
  </button>
</styled.button>
`;
